python-avd/pyavd/_eos_designs/structured_config/underlay/dhcp_server.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from functools import cached_property
+from ipaddress import AddressValueError, IPv4Address, ip_network
+
+from pyavd._utils import get
+
+from .utils import UtilsMixin
+
+
+class DhcpServerMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+
+    Class should only be used as Mixin to a AvdStructuredConfig class.
+    """
+
+    _hostvars: dict
+
+    @cached_property
+    def _subnets(self) -> list:
+        """
+        Returns a list of dhcp subnets for downstream p2p interfaces.
+
+        Used for l3 inband ztp/ztr.
+        """
+        subnets = []
+        for peer in self._avd_peers:
+            peer_facts = self.shared_utils.get_peer_facts(peer, required=True)
+            for uplink in peer_facts["uplinks"]:
+                if (
+                    uplink["peer"] == self.shared_utils.hostname
+                    and uplink["type"] == "underlay_p2p"
+                    and uplink.get("ip_address")
+                    and "unnumbered" not in uplink["ip_address"]
+                    and get(peer_facts, "inband_ztp")
+                ):
+                    subnet = {
+                        "subnet": str(ip_network(f"{uplink['peer_ip_address']}/{uplink['prefix_length']}", strict=False)),
+                        "ranges": [{"start": str(uplink["ip_address"]), "end": str(uplink["ip_address"])}],
+                        "name": f"inband ztp for {peer}-{uplink['interface']}",
+                        "default_gateway": f"{uplink['peer_ip_address'].split('/')[0]}",
```

Comment:
I'm not sure but regardless of if it is or isn't, since we're picking out the first item from the split operation (which will always be the IP address without the subnet), we'll wind up with the value we need.

Reply:
right, it's just that `uplink['peer_ip_address']` is used on lines 42 and 45 and line 42 is not expecting it to have a prefix length (no splitting there) but line 45 does.

Reply:
Oh I see.  Good point. I'll commit the suggestion.